Libraries/LibWeb/WebAssembly/Memory.cpp

```diff
@@ -100,8 +100,92 @@ WebIDL::ExceptionOr<u32> Memory::grow(u32 delta)
     return previous_size;
 }
 
+// https://webassembly.github.io/threads/js-api/index.html#dom-memory-tofixedlengthbuffer
+WebIDL::ExceptionOr<GC::Ref<JS::ArrayBuffer>> Memory::to_fixed_length_buffer()
+{
+    auto& vm = this->vm();
+
+    // 1. Let buffer be this.[[BufferObject]].
+    // 2. Let memaddr be this.[[Memory]].
+    // 3. If IsSharedArrayBuffer(buffer) is false,
+    if (m_shared == Shared::No) {
+        // 1. If IsFixedLengthArrayBuffer(buffer) is true, return buffer.
+        if (m_buffer && m_buffer->is_fixed_length())
+            return GC::Ref(*m_buffer);
+
+        // 2. Otherwise,
+        // 1. Let fixedBuffer be the result of creating a fixed length memory buffer from memaddr.
+        auto fixed_buffer = TRY(create_a_fixed_length_memory_buffer(vm, realm(), m_address, m_shared));
+
+        // 2. Perform ! DetachArrayBuffer(buffer, "WebAssembly.Memory").
+        if (m_buffer)
```

Comment:
:thinking: the spec seems to imply that `m_buffer` is never null?

Reply:
The existing code initialized `m_buffer` lazily upon first access to the JS `buffer` getter. I don't know the reason, and it's not exactly according to the spec. I think I'll have to make it not-lazy anyway, because the `buffer` getter in the threads JS API spec is actually a bit more complicated, and `m_buffer` being lazy would get in the way.